### ISSUE TITLE
`make assets` fails when repository not in $GOPATH

### ISSUE THREAD

**Issue by apghero** [prometheus/pushgateway#240]
## Bug Report
**What did you do?**

When attempting to recompile assets (`make assets`) it failed to work with my repository outside of the `$GOPATH`

**What did you expect to see?**

I expected to see assets compiled successfully.

**What did you see instead? Under which circumstances?**

```
 make assets                                                                                                           
>> writing assets                                                                                                                                               
2019/03/27 17:29:53 cannot find package "github.com/prometheus/pushgateway/resources" in any of:                                                                
        /usr/local/go/src/github.com/prometheus/pushgateway/resources (from $GOROOT)                                                                            
        /Users/agwozdziewycz/go/src/github.com/prometheus/pushgateway/resources (from $GOPATH)                                                                  
exit status 1                                                                                                                                                   
doc.go:22: running "go": exit status 1                                                                                                                          
make: *** [assets] Error 1                                                          
```

**Environment**

```
$ env | grep GO
GOPATH=/Users/agwozdziewycz/go
```

Nothing else should be relevant.

* System information:

		Darwin 18.2.0 x86_64

* Pushgateway version:

```
$ ./pushgateway --version
pushgateway, version 0.7.0 (branch: apghero/bugfix/190/route-prefix-bug, revision: 153da9be52b7851c67e552f7635ab6c91d209a60)
  build user:       agwozdziewycz@agwozdziew-ltm.internal.salesforce.com
  build date:       20190328-03:34:09
  go version:       go1.12.1
```

* Pushgateway command line:

		N/A.

**Comment by apghero** [comment]
Note that there is a workaround for this:

```
$ ln -s /Users/agwozdziewycz/Devel/pushgateway/resources/ ~/go/src/github.com/prometheus/pushgateway/resources
```

**Comment by beorn7** [comment]
I just tried to reproduce the problem in the following way:

```
$ mkdir not_in_GOPATH
$ cd not_in_GOPATH/
$ git clone git@github.com:prometheus/pushgateway
Cloning into 'pushgateway'...
remote: Enumerating objects: 24, done.
remote: Counting objects: 100% (24/24), done.
remote: Compressing objects: 100% (24/24), done.
remote: Total 2474 (delta 4), reused 14 (delta 0), pack-reused 2450
Receiving objects: 100% (2474/2474), 3.51 MiB | 6.52 MiB/s, done.
Resolving deltas: 100% (1392/1392), done.
$ cd pushgateway/
$ make assets
>> writing assets
writing assets_vfsdata.go
$ go version
go version go1.12.3 linux/amd64
$
```

I.e. it worked just fine. What Go version did you use?

**Comment by apghero** [comment]
```
$ go version
go version go1.12.1 darwin/amd64
```

Do you, by chance, have a standard GOPATH setup when you do that? So, does the following work?

```
$ mv ${GOPATH}{,.bak}                                                                                                           
$ cd /tmp && git clone git@github.com:prometheus/pushgateway                                                                    
Cloning into 'pushgateway'...                                                                                                                                   
remote: Enumerating objects: 30, done.
remote: Counting objects: 100% (30/30), done.
remote: Compressing objects: 100% (27/27), done.
remote: Total 2480 (delta 8), reused 20 (delta 3), pack-reused 2450
Receiving objects: 100% (2480/2480), 3.51 MiB | 1.08 MiB/s, done.
Resolving deltas: 100% (1396/1396), done.
$ cd pushgateway/
$ make assets
>> writing assets
go: finding github.com/alecthomas/template v0.0.0-20160405071501-a0175ee3bccc
go: finding github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf
go: finding github.com/prometheus/client_model v0.0.0-20180712105110-5c3871d89910                                                                              
go: finding github.com/sirupsen/logrus v1.2.0
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/common v0.0.0-20181126121408-4724e9255275
go: finding github.com/golang/protobuf v1.2.0
go: finding github.com/prometheus/client_golang v0.9.2
go: finding golang.org/x/tools v0.0.0-20181206194817-bcd4e47d0288
go: finding gopkg.in/alecthomas/kingpin.v2 v2.2.6
go: finding github.com/shurcooL/httpfs v0.0.0-20171119174359-809beceb2371
go: finding golang.org/x/crypto v0.0.0-20181203042331-505ab145d0a9
go: finding github.com/julienschmidt/httprouter v1.2.0
go: finding github.com/pmezard/go-difflib v1.0.0
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/shurcooL/vfsgen v0.0.0-20181202132449-6a9ea43bcacd
go: finding github.com/stretchr/objx v0.1.1
go: finding golang.org/x/crypto v0.0.0-20180904163835-0709b304e793
go: finding golang.org/x/net v0.0.0-20181201002055-351d144fa1fc
go: finding github.com/konsorten/go-windows-terminal-sequences v1.0.1
go: finding github.com/stretchr/testify v1.2.2
go: finding github.com/beorn7/perks v0.0.0-20180321164747-3a771d992973
go: finding github.com/prometheus/procfs v0.0.0-20181204211112-1dc9a6cbc91a
go: finding golang.org/x/sync v0.0.0-20181108010431-42b317875d0f
go: finding golang.org/x/sys v0.0.0-20181206074257-70b957f3b65e
go: finding golang.org/x/sys v0.0.0-20180905080454-ebe1bf3edb33
go: downloading github.com/shurcooL/vfsgen v0.0.0-20181202132449-6a9ea43bcacd
go: extracting github.com/shurcooL/vfsgen v0.0.0-20181202132449-6a9ea43bcacd
go: downloading github.com/shurcooL/httpfs v0.0.0-20171119174359-809beceb2371
go: extracting github.com/shurcooL/httpfs v0.0.0-20171119174359-809beceb2371
2019/04/11 15:26:09 cannot find package "github.com/prometheus/pushgateway/resources" in any of:                                                               
        /usr/local/go/src/github.com/prometheus/pushgateway/resources (from $GOROOT)                                                                           
        /Users/agwozdziewycz/go/src/github.com/prometheus/pushgateway/resources (from $GOPATH)                                                                 
exit status 1
doc.go:22: running "go": exit status 1
make: *** [assets] Error 1
$ mv ${GOPATH}{.bak,}
```

**Comment by beorn7** [comment]
I see. It works without any GOPATH, but with a set GOPATH but it will somehow try to use it, even with my modified Makefile in #243

**Comment by beorn7** [comment]
Got it. It's the `go run`call from the `go:generate` directive that somehow doesn't like the modules approach. Still looking...

**Comment by beorn7** [comment]
I think the problem is in our `asset/asset.go`. We are acting directly on `build.Import` there, and that cannot end well with modules.

**Comment by beorn7** [comment]
I think I have a fix, see #243 .
